src/cargo/util/network.rs

```diff
@@ -94,54 +97,97 @@ where
+        *config.shell() = Shell::from_write(Box::new(Vec::new()));
+        let result = with_retry(&config, || results.pop().unwrap());
+        assert!(result.is_ok())
     }
-    .into();
-    let error2 = HttpNot200 {
-        code: 502,
-        url: "Uri".to_string(),
+
+    #[test]
+    fn with_retry_finds_nested_spurious_errors() {
+        use crate::core::Shell;
+
+        //Error HTTP codes (5xx) are considered maybe_spurious and will prompt retry
+        //String error messages are not considered spurious
+        let error1 = anyhow::Error::from(HttpNot200 {
+            code: 501,
+            url: "Uri".to_string(),
+        });
+        let error1 = anyhow::Error::from(error1.context("A non-spurious wrapping err"));
+        let error2 = anyhow::Error::from(HttpNot200 {
+            code: 502,
+            url: "Uri".to_string(),
+        });
+        let error2 = anyhow::Error::from(error2.context("A second chained error"));
+        let mut results: Vec<CargoResult<()>> = vec![Ok(()), Err(error1), Err(error2)];
+        let config = Config::default().unwrap();
+        *config.shell() = Shell::from_write(Box::new(Vec::new()));
+        let result = with_retry(&config, || results.pop().unwrap());
+        assert!(result.is_ok())
     }
-    .into();
-    let mut results: Vec<CargoResult<()>> = vec![Ok(()), Err(error1), Err(error2)];
-    let config = Config::default().unwrap();
-    *config.shell() = Shell::from_write(Box::new(Vec::new()));
-    let result = with_retry(&config, || results.pop().unwrap());
-    assert!(result.is_ok())
-}
 
-#[test]
-fn with_retry_finds_nested_spurious_errors() {
-    use crate::core::Shell;
-
-    //Error HTTP codes (5xx) are considered maybe_spurious and will prompt retry
-    //String error messages are not considered spurious
-    let error1 = anyhow::Error::from(HttpNot200 {
-        code: 501,
-        url: "Uri".to_string(),
-    });
-    let error1 = anyhow::Error::from(error1.context("A non-spurious wrapping err"));
-    let error2 = anyhow::Error::from(HttpNot200 {
-        code: 502,
-        url: "Uri".to_string(),
-    });
-    let error2 = anyhow::Error::from(error2.context("A second chained error"));
-    let mut results: Vec<CargoResult<()>> = vec![Ok(()), Err(error1), Err(error2)];
-    let config = Config::default().unwrap();
-    *config.shell() = Shell::from_write(Box::new(Vec::new()));
-    let result = with_retry(&config, || results.pop().unwrap());
-    assert!(result.is_ok())
-}
+    #[test]
+    fn curle_http2_stream_is_spurious() {
+        let code = curl_sys::CURLE_HTTP2_STREAM;
+        let err = curl::Error::new(code);
+        assert!(maybe_spurious(&err.into()));
+    }
+
+    #[test]
+    fn git_cli_error_is_spurious() {
+        // This test is important because it also verifies that the regular
+        // expressions used to match git CLI transient errors are compilable,
+        // which is checked only at runtime.
```

Comment:
Unfortunately, git diff is not smart enough here.
The actual change is that tests were moved under `#[cfg(test)] mod tests` and this new test was added